docs: Improve README Structure and Content Clarity (#1448)
lambdaclass/lambda_ethereum_consensus: README.md

  | 🏗️   | [Phase 1: Beacon Chain Implementation](#phase-1-beacon-chain-implementation)     | Build the electra-upgraded beacon chain core         | • Apply electra-specific changes<br>• Run & pass full spec tests           | Run spec suite (`make spec-test`), aim for 0 failures |
  | ⌛   | [Phase 2: P2P & Sepolia Long-Running Sessions](#phase-2-p2p--sepolia-long-running-sessions)   | Ensure stability on Sepolia                          | • Implement the P2P changes <br> • Deploy the node on our server pointing to Sepolia<br>• Fix every issue we found that interrupts the node execution | Continuous uptime checks & up-to-date block processing for 72+ hrs in Sepolia|
  | ⌛   | [Phase 3: Validator Upgrades](#phase-3-validator-upgrades) | Ensure validators duties on devnets              |• Implement the honest validator changes<br>• Make assertoor work<br> • Test via Kurtosis & Assertoor | Execute Kurtosis scenarios & Assertoor with continuous uptime checks and up-to-date validation duties for 72+ hrs on kurtosis   |

  For more details, see the [Implementation Gaps for electra Upgrade](./electra-gap.md).

>   ### Deneb and previous forks support
>   We prioritize Electra support over being 100% compatible with previous forks. Some issues will be address while making Electra work, but we can present the current status as follows: 

Reviewer: We don't currently support previous forks, this was a decision made from the beginning as far as I understand. This was decided in the first migration since the client started (from capella to deneb). As i understood it the idea is to achieve a complete working version of the last fork and start supporting fork update from that onwards. So that being said i think this section could instead reflect the road to the MVP:

```suggestion
  ### Road to the MVP
  Once an initial version of electra is in place we'll need to work on some missing component before reaching the MVP state: 
```

>   #### Without Validators
>   - [✅] Sync Sepolia in Daneb.
>   - [🏗️] Implementation of the Beacon API
>   - [  ] Improve performance to process blocks and epochs for other networks

Reviewer: Here we should talk probably more about the actual goal isntead of the fork given the changes in the previous comment

```suggestion
  #### Without Validators
  - [✅] Sync and run the node on Sepolia for validating state transitions 
  - [🏗️] Implementation of the Beacon API
  - [  ] Improve performance to process blocks and epochs for other networks (specially Holesky/Hoodi/Mainnet)
```

  #### With Validators
  - [🏗️] Run devnet using kurtosis being validator

  ## Security

  We take security seriously. If you discover a vulnerability, please report it responsibly:

  - Use the **[GitHub "Report a Vulnerability" feature](../../security/advisories/new)**.
  - Alternatively, email **[security@lambdaclass.com](mailto:security@lambdaclass.com)**.

  Refer to our [Security Policy](./.github/SECURITY.md) for more details.

  ## Getting Started

  ### Prerequisites

  Install the following tools:

  - [Git](https://git-scm.com/)
  - [wget](https://www.gnu.org/software/wget/)
  - [CMake](https://cmake.org/)
  - [Elixir](https://elixir-lang.org/install.html)
  - [Erlang](https://www.erlang.org/downloads)
  - [Go](https://go.dev/doc/install)
  - [Rust](https://www.rust-lang.org/tools/install)
  - [Protoc](https://grpc.io/docs/protoc-installation/)

  You can install the necessary components directly from official sources or alternatively, use **asdf** for version management. 

>   See [Prerequisites](./docs/INSTALLATION.md#prerequisites) for detailed instructions.

Reviewer: I guess it's the prerequesites MD now
```suggestion
  See [Prerequisites](./docs/PREREQUISITES.md) for detailed instructions.
```

  - [Protoc](https://grpc.io/docs/protoc-installation/)

  You can install the necessary components directly from official sources or alternatively, use **asdf** for version management. 

  See [Prerequisites](./docs/INSTALLATION.md#prerequisites) for detailed instructions.

  ## Installing and running

  There are Makefile targets for these tasks.

  > [!TIP]
  > You can list the available targets with `make help`

  > [!NOTE]
  > If `make deps` is failing with `protoc-gen-go: program not found or is not executable` you might need to run
  > ```Shell
  > export PATH=$PATH:~/go/bin
  > ```

  ```shell
  make deps # Installs dependencies
  make iex  # Runs a terminal with the application started
  ```

  The iex terminal can be closed by pressing ctrl+c two times.

  > [!WARNING]
  > The node isn't capable of syncing from genesis yet, and so requires using checkpoint-sync to start (see [Checkpoint Sync](#checkpoint-sync)).
  > In case checkpoint-sync is needed, `make iex` will end immediately with an error.

  For more details about installing dependecies, see [Installing](./docs/PREREQUISITES.md).

  ## Usage

  Here you'll find both basic and advanced usage examples.  

>   For implementation details, refer to the [Architecture](./docs/ARCHITECTURE.md) documentation.

Reviewer: The current MD is not in mayus:

```suggestion
  For implementation details, refer to the [Architecture](./docs/architecture.md) documentation.
```

Or we need to rename it if we want to maintain consistency with TESTING and PREREQUISITES although most of the other MDs are not in mayus either.
